src/services/marketsreq.ts

```diff
@@ -1,7 +1,25 @@
-import type { ZZServiceHandler } from 'src/types'
+import type { ZZMarketInfo, ZZServiceHandler } from 'src/types'
+
+export const marketsreq: ZZServiceHandler = async (
+  api, 
+  ws,
+  [chainId, detailedFlag]
+) => {
+  let marketsMsg
+  if (detailedFlag) {
+    const marketInfo: ZZMarketInfo[] = []
+    const activeMarkets = await api.redis.SMEMBERS(`activemarkets:${chainId}`)
+    const result = activeMarkets.map(async (market: string) => {
+      const details: ZZMarketInfo = await api.getMarketInfo(market, chainId)
+      if (details) marketInfo.push(details)
+    })
+    Promise.all(result)
```

Comment:
I think you need an await here

Reply:
added with https://github.com/ZigZagExchange/backend/pull/43/commits/b92a86e0a9f22bbc3f06803f001936eedd75353a